### PR TITLE
Avoid NPE in testThreadMXBeanProxy debug output

### PR DESCRIPTION
**Pull request by pshipton**
Related to https://github.com/eclipse-openj9/openj9/pull/19505
Issue https://github.com/eclipse-openj9/openj9/issues/18920

### CODE REVIEW

File: test/functional/JLM_Tests/src/org/openj9/test/java/lang/management/TestManagementFactory.java [hunk 1]
```diff
@@ -1316,7 +1316,7 @@ public void testThreadMXBeanProxy() {
 				System.out.println("lockOwnerName ti1: " + ti1.getLockOwnerName() + " ti2: " + ti2.getLockOwnerName());
 				LockInfo li1 = ti1.getLockInfo();
 				LockInfo li2 = ti2.getLockInfo();
-				if (!li2.equals(li1)) {
+				if (li2 != li1) {
```
Comment (keithc-ca) [inline test/functional/JLM_Tests/src/org/openj9/test/java/lang/management/TestManagementFactory.java:1319]: It seems `LockInfo` inherits `equals()` from `Object`, so this seems ok, but perhaps a safer solution is to use `Objects.equals()`.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: test/functional/JLM_Tests/src/org/openj9/test/java/lang/management/TestManagementFactory.java (hunk 1 above)
Comment (pshipton) [reply test/functional/JLM_Tests/src/org/openj9/test/java/lang/management/TestManagementFactory.java]: Updated.

**Review by keithc-ca — APPROVED** [review]

**Comment by keithc-ca** [comment]
Jenkins test sanity alinux jdk17

**Comment by keithc-ca** [comment]
Jenkins test sanity alinux jdk17

**Pull request merged**